### PR TITLE
python3Packages.plantuml: init at 0.3.0

### PR DESCRIPTION
**Pull request by nikstur**
###### Description of changes

New package:
python-plantuml is a Plantuml HTTP client.
https://github.com/dougn/python-plantuml

I need this package primarily as a dependency for [plantuml-markdown](https://github.com/mikitex70/plantuml-markdown/blob/master/setup.py). I plan to open a PR for plantuml-markdown after this package has been merged. 

Please let me know if declaring the actual plantuml application as `plantumlBinary` is the nix idiomatic way of doing things.

`plantuml/default.nix` has been formatted with nixpkgs-fmt and checked with statix

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by smancill** [comment]
Instead of `plantumlBinary` you just pass `pkgs`. See the second example in https://nixos.org/manual/nixpkgs/stable/#handling-dependencies

**Comment by nikstur** [comment]
Thanks for the pointer @smancill! Does it look fine to you now?

**Comment by smancill** [comment]
`twine` is not needed, since it is for publishing to PyPI. `wheel` is also not needed, it's not listed in `setup.py`.

And it doesn't look like you are building the docs, so neither is `sphinx` (and `sphinx` should go into `nativeBuildinputs` in that case).

File: pkgs/development/python-modules/plantuml/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  propagatedBuildInputs = [
+    sphinx
+    httplib2
+    twine
+    wheel
+    six
+    pkgs.plantuml
+  ];
+
+  doCheck = false;
```
Comment (smancill) [inline pkgs/development/python-modules/plantuml/default.nix:33]: It's good idea to put a comment of why not running tests.

File: pkgs/development/python-modules/plantuml/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    sha256 = "sha256-aPXPqoKlu8VLi0Jn84brG7v3qM9L18Ut4sabYYGb3qQ=";
+  };
+
+  propagatedBuildInputs = [
+    sphinx
+    httplib2
+    twine
+    wheel
+    six
+    pkgs.plantuml
```
Comment (smancill) [inline pkgs/development/python-modules/plantuml/default.nix:30]: Reading the code, locally installed `plantuml` is never really used, it connects to a server, so I don't think this qualifies as a dependency.

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -6703,6 +6703,8 @@ in {
 
   plaid-python = callPackage ../development/python-modules/plaid-python { };
 
+  plantuml = callPackage ../development/python-modules/plantuml { };
+
   plaster = callPackage ../development/python-modules/plaster { };
```
Comment (smancill) [inline pkgs/top-level/python-packages.nix:6708]: Looks like you'll need update `master` and rebase, it looks like there is a conflict in this part.

**Comment by nikstur** [comment]
> `twine` is not needed, since it is for publishing to PyPI. `wheel` is also not needed, it's not listed in `setup.py`.
> 
> And it doesn't look like you are building the docs, so neither is `sphinx` (and `sphinx` should go into `nativeBuildinputs` in that case).

You're right. I just blindly copied the contents of the requirements.txt without thinking about whether these are actually required. Thanks for your time in reviewing this. 

I hope force pushing the changes was alright. My understanding is that in the end the commit message should say directly what was achieved instead of  having multiple commits list me fixing my mistakes. :)

**Review by smancill — APPROVED** [review]

**Pull request merged**